taskotron_python_versions/requires.py

```diff
@@ -14,6 +14,29 @@
 INFO_URL = 'https://pagure.io/packaging-committee/issue/686'
 
 
+class DNFQuery(object):
+
+    """DNF Qeuery API.
+
+    Initializes the query only when needed
+    and saves it for reuse.
+    """
+
+    def __init__(self, release):
+        self.release = release
+        self.query = None
+
+    def get_packages_by(self, **kwargs):
+        """Return the result of the DNF query execution,
+        filtered by kwargs.
+        """
+        if not self.query:
+            self.query = get_dnf_query(self.release)
+            if not self.query:
```

Comment:
> This only happens if there was an error? 

Yes.

> If so, I think it's Ok that get_dnf_query() will be called again for next get_packages_by() call.

Yes, it will be called again in the next `get_packages_by() `.